samples/android/Continote/app/src/main/java/com/firebasecontinue/sample/continote/NoteListItemViewHolder.java

```diff
@@ -0,0 +1,116 @@
+            throw new AssertionError("itemView, note, and database key must be non-null/non-empty");
+        }
+
+        update(itemView, note, databaseKey);
+    }
+
+    /**
+     * Updates this ViewHolder and the UI of the View managed by it based on the provided Note and
+     * database key.
+     *
+     * @param itemView The View managed by this ViewHolder which may need its UI updated.
+     * @param note The Note this ViewHolder is for (to update the UI based on).
+     * @param databaseKey The database key of the Note this ViewHolder is for.
+     */
+    public void update(@NonNull View itemView, @NonNull Note note, @NonNull String databaseKey) {
+        if (itemView == null || note == null || databaseKey == null || databaseKey.length() == 0) {
+            // This should not happen, but just in case.
+            throw new AssertionError("itemView, note, and database key must be non-null/non-empty");
+        }
+
+        mNoteDatabaseKey = databaseKey;
+
+        TextView titleTextView = (TextView) itemView.findViewById(R.id.noteItemTitleTextView);
+        setTextWithPlaceholder(titleTextView, note.getTitle(), "No Title");
+        TextView contentTextView = (TextView) itemView.findViewById(R.id.noteItemContentTextView);
+        setTextWithPlaceholder(contentTextView, note.getContent(), "No Content");
+    }
+
+    /**
+     * Gets and returns Note database key.
+     *
+     * @return The Firebase Realtime Database key for the Note this ViewHolder is for.
+     */
+    @Nullable
+    public String getNoteDatabaseKey() {
+        return mNoteDatabaseKey;
+    }
+
+    /**
+     * Sets the text of the provided textView to the provided value, or the placeholder if the
+     * value is null or empty. Also italicizes the textView if the placeholder is used.
+     *
+     * This could go into a "Utils" class, but this is the only place it is used, so this
+     * is sufficient for this sample app.
+     *
+     * @param textView The TextView to set the text for.
+     * @param value The value to set the text to.
+     * @param placeholder The placeholder text to use if the value above is null or empty.
+     */
+    private static void setTextWithPlaceholder(@NonNull TextView textView,
+                                               @Nullable String value,
+                                               @Nullable String placeholder) {
+        if (textView == null) {
+            // This should not happen, but just in case.
+            throw new AssertionError("textView must be non-null");
+        }
+
+        if (value != null && value.length() > 0) {
+            // The provided value is nonempty, so use it.
+            textView.setText(value);
+            textView.setTypeface(textView.getTypeface(), Typeface.NORMAL);
+        } else {
+            // The provided value is empty, so use the placeholder.
+            textView.setText(placeholder);
+            textView.setTypeface(textView.getTypeface(), Typeface.ITALIC);
```

Comment:
Thanks for the suggestion. I'll leave this as a possible improvement in the future, since this placeholder logic is only used in one place anyways (the ListView items).